File: sync/v1/sync_map_item/update-update-map-item/update-update-map-item.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
# DANGER! This is insecure. See http://twil.io/secure
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

sync_map_item = @client.sync
                       .services('ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
                       .sync_maps('Players')
                       .sync_map_items('steph_curry')
                       .update(data: {
                            'name' => 'Stephen Curry',
                            'level' => 31,
                            'username' => 'spicy_curry'
                        })

puts sync_map_item.key

File: sync/v1/sync_map_item/update-update-map-item/update-update-map-item.5.x.php
<?php

// Update the path below to your autoload.php,
// see https://getcomposer.org/doc/01-basic-usage.md
require_once '/path/to/vendor/autoload.php';

use Twilio\Rest\Client;

// Your Account Sid and Auth Token from twilio.com/console
$sid    = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
$token  = "your_auth_token";
$twilio = new Client($sid, $token);

$sync_map_item = $twilio->sync->v1->services("ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
                                  ->syncMaps("MPXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
                                  ->syncMapItems("steph_curry")
                                  ->update(array(
                                               "data" => array(
                                                   "name" => "Stephen Curry",
                                                   "level" => "31",
                                                   "username" => "spicy_curry"
                                               )
                                           )
                                  );

print($sync_map_item->key);